src/extension.ts

```diff
@@ -70,10 +73,26 @@ export function activate(context: vscode.ExtensionContext): void {
 			}
 		},
 	);
+
+	const selectEditModeCommand = vscode.commands.registerCommand(
+		"hexEditor.selectEditMode",
+		async () => {
+			const pickEditMode = await vscode.window.showQuickPick(["Insert", "Replace"]);
+			const messaging = registry.activeMessaging[Symbol.iterator]().next().value as ExtensionHostMessageHandler | undefined;
+			if(messaging && pickEditMode && registry.activeDocument) {
+				const mode = pickEditMode === "Insert" ? HexDocumentEditOp.Insert : HexDocumentEditOp.Replace;
+				messaging.sendEvent({ type: MessageType.SetEditMode, mode: mode});
```

Comment:
I would instead have a setter on the document that triggers an `onDidChangeEditMode` event that is listened to in the `HexEditorProvider`. You'll also want to have the HexEditorProvider seed the initial document state when the webview is created to handle restoration.